bin/add_frontend_server.rb

```diff
@@ -0,0 +1,120 @@
+require 'aws-sdk'
+require_relative '../deployment'
+
+MAX_WAIT_TIME = 600 #Wait no longer than 10 minutes for instance creation. Typically this takes a couple minutes
+
+Aws.config.update({
+                      region: CDO.aws_region,
+                      credentials: Aws::Credentials.new(CDO.aws_access_key, CDO.aws_secret_key),
+                  })
+
+ec2client = Aws::EC2::Client.new
+
+instances = ec2client.describe_instances
+
+#Determine distribution of availability zones, pick the one that has the least capacity among frontend instances
+instance_distribution = Hash.new(0)
+
+instances.reservations.each do |reservation|
+  reservation.instances.each do |instance|
+    unless instance.state.name == 'running'
+      next
+    end
+
+    is_frontend = true
+
+    instance.tags.each do |tag|
```

Comment:
I think our styleguide has us avoiding `and` (because it can have surprising behavior around not actually being the same as `&&`)